observable-slim.js

```diff
@@ -191,7 +191,20 @@ var ObservableSlim = (function() {
 				
 				// record the deletion that just took place
 				changes.push({"type":"delete","target":target,"property":property,"newValue":null,"previousValue":previousValue[property],"currentPath":currentPath,"proxy":proxy});
-				
+
+                var t = targets.indexOf(target);
```

Comment:
It looks like your objective is to use `beforeChange` to interrupt or block changes from happening (as opposed to modifying the values before they are set). If we add `beforeChange` functionality, then we also have to think about supporting the ability to modify values before they are set (a very common use case for ES6 Proxy). Once you start thinking about that, then you can see how fraught it becomes to loop over all observables for a given target object.

I think it'd be more appropriate to only trigger the `beforeChange` callback for the observable through which the property was modified. You can access `observable.beforeChange` directly from here (without having to loop). That will significantly boost performance. I'm currently building out support for IE11 with the Google Chrome Proxy polyfill and IE11 already struggles with large objects.